### ISSUE TITLE
Wezterm does not have window decorations/titlebar on Wayland

### ISSUE THREAD

**Issue by rishubn** [wez/wezterm#4962]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Mutter

### WezTerm version

20240204-213233-b03b833a

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

(Ubuntu 22.04) On wayland, wezterm spawns without window decorations. It  cannot be resized or dragged with the mouse like other windows. This does not appear when I log in with X11.

Below is a gtk app next to wezterm, notice the missing header
![image](https://github.com/wez/wezterm/assets/6591138/e78df84f-2a65-4fd4-9c80-a07c7504f112)

### Configuration

```lua
rnagpal@ipn078:~$ cat .config/wezterm/wezterm.lua
-- Pull in the wezterm API
local wezterm = require 'wezterm'

-- This table will hold the configuration.
local config = {}

-- In newer versions of wezterm, use the config_builder which will
-- help provide clearer error messages
if wezterm.config_builder then
  config = wezterm.config_builder()
end

-- This is where you actually apply your config choices

-- For example, changing the color scheme:
config.color_scheme = 'tokyonight-storm'
config.enable_scroll_bar = true
config.window_background_opacity = 0.9
config.default_cursor_style = "SteadyBar"
config.window_decorations = "TITLE | RESIZE"
-- and finally, return the configuration to wezterm

return config
```

### Expected Behavior

Have the same style and functionality as other windows in wayland

**Comment by wez** [comment]
Wayland support is currently being re-implemented in `main`, this is an outstanding issue around that

**Comment by codewiz** [comment]
Still seeing this bug with the GNOME 47 Wayland session. Using Wezterm built today from git head.

**Comment by sandikata** [comment]
Same issue for me.

Gentoo, Gnome 47, Mutter 47 -> Wayland

Latest git build

**Comment by cguentherTUChemnitz** [comment]
> Had the same problem under Fedora 40. As a workaround I "disabled wayland" in wezterm's config.
> 
> `config.enable_wayland = false `
> 
> Hope this helps for now

Sadly this workaround is a no go on high-dpi displays with fractional scaling in wayland. The enable_wayland just produces badly blurred font rendering in this scenario.

**Comment by hnjae** [comment]
> > Had the same problem under Fedora 40. As a workaround I "disabled wayland" in wezterm's config.
> > `config.enable_wayland = false `
> > Hope this helps for now
> 
> Sadly this workaround is a no go on high-dpi displays with fractional scaling in wayland. e.g. in gnome. The enable_wayland just produces badly blurred font rendering in this scenario.

@cguentherTUChemnitz You can enable 'xwayland-native-scaling' in that case, which stops GNOME from scaling Xwayland windows. Apply the following with dconf-editor or gsettings:

```
[org/gnome/mutter]
experimental-features=['xwayland-native-scaling']
```

**Comment by davidgroves** [comment]
Still a problem on Debian-Unstable with Wayland and Mutter 48.1, with Wezterm built from the git tree on the same day as this comment.

**Comment by valpackett** [comment]
Will be fixed in my PR adopting sctk-adwaita, but the bug is basically invoked by `window_decorations` being set to `TITLE | RESIZE` (which is the default). In that configuration, current code would ask for server-side decorations and just assume they exist, without confirming that they do.

You can use `window_decorations = "INTEGRATED_BUTTONS | RESIZE"` fine even right now.

**Comment by valpackett** [comment]
@gkmcd you're just declaring `window_decorations` as a variable floating around, not the property on `config.`

**Comment by gkmcd** [comment]
@valpackett whoops, thanks. Now I have the integrated buttons and the little titlebar others have reported.

![Image](https://github.com/user-attachments/assets/8e371ba1-7824-4c55-a6c5-b3b32221bb88)

**Comment by valpackett** [comment]
right, current git version has a very busted frame, see #6923 for a fix

**Comment by ahas** [comment]
On `6.18.7-200.fc43.x86_64` and `wezterm 20260201_005028_2672d19e`, when `enable_wayland = true`, there is no title bar.
If `config.window_decorations = "INTEGRATED_BUTTONS"`, the buttons appear, but dragging the window via the title bar still doesn’t work.

**Comment by mati865** [comment]
@ahas you need changes from https://github.com/wezterm/wezterm/pull/7095 to make it work.

**Comment by BrandonJF** [comment]
Still happening on current nightly (`20260912-133823-2afb8364`), so this isn't fixed as of Sept 2026. Ubuntu 26.04, GNOME/Mutter 50.1, Wayland.

`window_decorations = "RESIZE|TITLE"` gives no title bar, nothing to drag by. Window itself works fine otherwise — Alt+F8 resize works, rendering's fine, it's just the decoration/drag handling that's broken.

Might help narrow it down: same binary, same config, just unset `WAYLAND_DISPLAY` so it falls back to XWayland, and the title bar shows up and drags normally. So it seems isolated to wezterm's Wayland decoration/move handling specifically, not something Mutter's doing wrong on its end.

Thanks for the hard work and attention here devs!
